src/generalVis/components/GeneralHome.tsx

```diff
@@ -0,0 +1,265 @@
+    const [currentVis, setCurrentVis] = useState<ESupportedPlotlyVis>(ESupportedPlotlyVis.SCATTER);
+    const [selectedCatCols, setSelectedCatCols] = useState<string[]>(
+        props.columns.filter((c) => c.selectedForMultiples === true && c.type === EColumnTypes.CATEGORICAL).map((c) => c.name)
+    );
+    const [selectedNumCols, setSelectedNumCols] = useState<string[]>(
+        props.columns.filter((c) => c.selectedForMultiples === true && c.type === EColumnTypes.NUMERICAL).map((c) => c.name)
+    );
+
+    const [bubbleSize, setBubbleSize] = useState<NumericalColumn | null>(null);
+    const [colorMapping, setColorMapping] = useState<CategoricalColumn | null>(null);
+    const [opacity, setOpacity] = useState<NumericalColumn | null>(null);
+    const [shape, setShape] = useState<CategoricalColumn | null>(null);
+    const [barGroup, setBarGroup] = useState<CategoricalColumn | null>(null);
+    const [barMultiples, setBarMultiples] = useState<CategoricalColumn | null>(null);
+    const [barDisplayType, setBarDisplayType] = useState<EBarDisplayType>(EBarDisplayType.DEFAULT);
+    const [barGroupType, setBarGroupType] = useState<EBarGroupingType>(EBarGroupingType.STACK);
+    const [barDirection, setBarDirection] = useState<EBarDirection>(EBarDirection.VERTICAL);
+
+    const updateBubbleSize = (newCol: string) => setBubbleSize(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.NUMERICAL)[0] as NumericalColumn);
+    const updateOpacity = (newCol: string) => setOpacity(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.NUMERICAL)[0] as NumericalColumn);
+    const updateColor = (newCol: string) => setColorMapping(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.CATEGORICAL)[0] as CategoricalColumn);
+    const updateShape = (newCol: string) => setShape(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.CATEGORICAL)[0] as CategoricalColumn);
+    const updateBarGroup = (newCol: string) => setBarGroup(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.CATEGORICAL)[0] as CategoricalColumn);
+    const updateBarMultiples = (newCol: string) => setBarMultiples(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.CATEGORICAL)[0] as CategoricalColumn);
+
+    const updateCurrentVis = (s: ESupportedPlotlyVis) => setCurrentVis(s);
+    const updateSelectedCatCols = (s: string[]) => setSelectedCatCols(s);
+    const updateSelectedNumCols = (s: string[]) => setSelectedNumCols(s);
+
+    const updateBarDisplayType = (s: EBarDisplayType) => setBarDisplayType(s);
+    const updateBarGroupType = (s: EBarGroupingType) => setBarGroupType(s);
+    const updateBarDirection = (s: EBarDirection) => setBarDirection(s);
+
+    const shapeScale = useMemo(() => {
+        return shape ?
+            scale.ordinal<string>().domain(d3.set(shape.vals.map((v) => v.val)).values()).range(['circle', 'square', 'triangle-up', 'star'])
```

Comment:
```suggestion
            d3.scale.ordinal<string>().domain(new Set(shape.vals.map((v) => v.val)).values()).range(['circle', 'square', 'triangle-up', 'star'])
```

Can we use a regular `Set` here and avoid D3?